src/main/scala/nl.knaw.dans.easy.bagstore/server/BagsServletComponent.scala

```diff
@@ -60,5 +61,29 @@ trait BagsServletComponent extends DebugEnhancedLogging {
             InternalServerError(s"[${ new DateTime() }] Unexpected type of failure. Please consult the logs")
         }
     }
+
+    get("/:uuid/*") {
+      val uuidStr = params("uuid")
+      multiParams("splat") match {
+        case Seq(path) =>
+          ItemId.fromString(s"""$uuidStr/${ path }""")
+            .recoverWith {
+              case _: IllegalArgumentException => Failure(new IllegalArgumentException(s"invalid UUID string: $uuidStr"))
+            }
+            .flatMap(itemId => {
+              debug(s"Retrieving item $itemId")
+              bagStores.getStream(itemId, response.outputStream)
+            })
+            .map(_ => Ok())
+            .getOrRecover {
+              case e: IllegalArgumentException => BadRequest(e.getMessage)
+              case e: NoSuchBagException => NotFound(e.getMessage)
+              case e: NoSuchFileException => NotFound(e.getMessage)
+              case NonFatal(e) =>
+                logger.error("Error retrieving bag", e)
+                InternalServerError(s"[${ new DateTime() }] Unexpected type of failure. Please consult the logs")
```

Comment:
DRY, extract the following methods:


    private def illegalArgument(uuidStr: String) = {
      Failure(new IllegalArgumentException(s"invalid UUID string: $uuidStr"))
    }

    private def unexpectedFailure(message: String, e: Throwable) = {
      logger.error(message, e)
      InternalServerError(s"[${ new DateTime() }] Unexpected type of failure. Please consult the logs")
    }